xwrf/grid.py

```diff
@@ -0,0 +1,120 @@
+"""Provide grid-related functionality specific to WRF datasets.
+
+This submodule contains code reused with modification from Salem under the terms of the BSD
+3-Clause License. Salem is Copyright (c) 2014-2021, Fabien Maussion and Salem Development Team All
+rights reserved.
+"""
+
+import numpy as np
+import pyproj
+
+# Default CRS (lon/lat on WGS84, which is EPSG:4326)
+wgs84 = pyproj.CRS(4326)
+
+# WRF dimension names
+_HORIZONTAL_DIMS = ['south_north', 'west_east', 'south_north_stag', 'west_east_stag']
+
+
+def _wrf_grid_from_dataset(ds):
+    """Get the WRF projection and dimension coordinates out of the file."""
+
+    pargs = {'x_0': 0, 'y_0': 0, 'a': 6370000, 'b': 6370000}
+
+    # Use standards from a typical WRF file
+    cen_lon = ds.CEN_LON
+    cen_lat = ds.CEN_LAT
+    dx = ds.DX
+    dy = ds.DY
+    pargs['lat_1'] = ds.TRUELAT1
+    pargs['lat_2'] = getattr(ds, 'TRUELAT2', ds.TRUELAT1)
+    pargs['lat_0'] = ds.MOAD_CEN_LAT
+    pargs['lon_0'] = ds.STAND_LON
+    pargs['center_lon'] = ds.CEN_LON
+    proj_id = ds.MAP_PROJ
```

Comment:
Could we merge the dictionary assignment?

```suggestion

    # Use standards from a typical WRF file
    cen_lon = ds.CEN_LON
    cen_lat = ds.CEN_LAT
    dx = ds.DX
    dy = ds.DY
    pargs = {
        'x_0': 0,
        'y_0': 0,
        'a': 6370000,
        'b': 6370000,
        'lat_1': ds.TRUELAT1,
        'lat_2': getattr(ds, 'TRUELAT2', ds.TRUELAT1),
        'lat_0': ds.MOAD_CEN_LAT,
        'lon_0': ds.STAND_LON,
        'center_lon': ds.CEN_LON,
    }

    proj_id = ds.MAP_PROJ
```

Reply:
Yes, good catch now that we don't have the conditional around any more!